lib/ui/painting/image_encoding.cc

```diff
@@ -44,11 +51,17 @@ void InvokeDataCallback(std::unique_ptr<DartPersistentValue> callback,
   tonic::DartState::Scope scope(dart_state);
   if (!buffer) {
     DartInvoke(callback->value(), {Dart_Null()});
-  } else {
-    Dart_Handle dart_data = tonic::DartConverter<tonic::Uint8List>::ToDart(
-        buffer->bytes(), buffer->size());
-    DartInvoke(callback->value(), {dart_data});
+    return;
   }
+  FML_DCHECK(buffer->unique());
```

Comment:
`SkData` buffers are immutable.  It should be safe to wrap the `SkData` in a Dart external typed data even if someone else is also holding a reference to the `SkData`.

Reply:
The `writable_data()` getter below will do a uniqueness check. I opted to hoist it out to here with a comment.

Reply:
Skia will never write into a clients SkData buffer (I'll CC you the clarification the Skia folks made about this). I am not sure why you need to invoke `writable_data()` at all though. The docs for `Dart_NewExternalTypedDataWithFinalizer` say "A data array. This array must not move." for the `data` argument. So it seems like Dart won't write to it either. In that case, call `data()` on `SkData` and `const_cast` it to to suit the requirements of the Dart API (I am not sure why that API does not say `const void *` but maybe you do.

Reply:
Ah, okay. The external Uint8List is generally writable, but I can wrap it in a `UnmodifiableUint8ListView` ([here](https://api.dart.dev/stable/2.8.4/dart-typed_data/UnmodifiableUint8ListView-class.html)) before returning it from the runtime call. I'll do that.

Reply:
Done. ptal.